images/hub/Dockerfile

```diff
@@ -72,3 +72,36 @@ USER ${NB_USER}
 EXPOSE 8081
 ENTRYPOINT ["tini", "--"]
 CMD ["jupyterhub", "--config", "/usr/local/etc/jupyterhub/jupyterhub_config.py"]
+
+
+# The final stage - default version
+# ---------------------------------
+# We add a few non-critical packages on top of the slim version to provide some
+# additional utility.
+#
+FROM slim-stage as default-stage
+
+USER root
+
+ARG PIP_CACHE_DIR=/tmp/pip-cache
+RUN --mount=type=cache,target=${PIP_CACHE_DIR} \
+    --mount=type=cache,from=build-stage,source=/tmp/wheels,target=/tmp/wheels \
+    pip install \
+        --find-links=/tmp/wheels/ \
+        # Updates below should be repeated in build-stage.
+        #
+        # py-spy is useful for profiling performance of running hubs
+        py-spy
+
+RUN apt-get update \
+ && apt-get install -y --no-install-recommends \
+        # misc network utilities
```

Comment:
```suggestion
```
`git` isn't a network utility. I think all these are common enough that the comments can be dropped (and if someone hasn't heard of `less ` or `vim` then `misc other utilities` isn't very informative).